docs/subnets/case-study-wagmi-upgrade.md

```diff
@@ -0,0 +1,158 @@
+
+With the above `upgrade.json`, we intend to change the `adminAddresses` at timestamp `1660658400`:
+
+- `0x6f0f6DA1852857d7789f68a28bba866671f3880D` is named as the new Admin of the FeeManager
+- `1660658400` is the [Unix timestamp](https://www.unixtimestamp.com/) for 10:00 AM EDT 08/16/2022 (future time when we made the announcement) when the new FeeManager change would take effect.
+
+We place the `upgrade.json` file in the chain config directory, which in our case is `~/.avalanchego/configs/chains/2ebCneCbwthjQ1rYT41nhd7M76Hc6YmosMAQrTFhBq8qeqh6tt/`. After that, we restart the node so the upgrade file is loaded.
+
+When the node restarts, AvalancheGo reads the contents of the JSON file and passes it into Subnet-EVM. We see a log of the chain configuration that includes the updated precompile upgrade. It looks like this:
+
+```text
+INFO [08-15|15:09:36.772] <2ebCneCbwthjQ1rYT41nhd7M76Hc6YmosMAQrTFhBq8qeqh6tt Chain>
+github.com/ava-labs/subnet-evm/eth/backend.go:155: Initialised chain configuration
+config=“{ChainID: 11111 Homestead: 0 EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0
+Constantinople: 0 Petersburg: 0 Istanbul: 0, Muir Glacier: 0, Subnet EVM: 0, FeeConfig:
+{\“gasLimit\“:20000000,\“targetBlockRate\“:2,\“minBaseFee\“:1000000000,\“targetGas\
+“:100000000,\“baseFeeChangeDenominator\“:48,\“minBlockGasCost\“:0,\“maxBlockGasCost\
+“:10000000,\“blockGasCostStep\“:500000}, AllowFeeRecipients: false, NetworkUpgrades: {\
+“subnetEVMTimestamp\“:0}, PrecompileUpgrade: {}, UpgradeConfig: {\“precompileUpgrades\“:
+[{\“feeManagerConfig\“:{\“adminAddresses\“:[\
+“0x6f0f6da1852857d7789f68a28bba866671f3880d\“],\“blockTimestamp\“:1660658400}}]},
+Engine: Dummy Consensus Engine}”
+```
+
+We note that `precompileUpgrades` correctly shows the upcoming precompile upgrade. Upgrade is locked in and ready.
+
+## Activation
+
+When the time passed 10:00 AM EDT 08/16/2022 (unix timestamp 1660658400), the `upgrade.json` had been executed as planned and the new FeeManager admin address has been activated. From now on, we don't need to issue any new code or deploy anything on the WAGMI nodes to change the fee structure. Let's see how it works in practice!
+
+## Using Fee Manager
+
+The owner `0x6f0f6DA1852857d7789f68a28bba866671f3880D` can now configure the fees on the Subnet as they see fit. To do that, all that's needed is access to the network, the private key for the newly set manager address and making calls on the precompiled contract.
+
+We will use [Remix](https://remix.ethereum.org) online Solidity IDE and the [Core Browser Extension](https://support.avax.network/en/articles/6066879-core-extension-how-do-i-add-the-core-extension). Core comes with WAGMI network built-in. Metamask will do as well but you will need to [add WAGMI](./wagmi.md#adding-wagmi-to-metamask) yourself.
+
+First using Core, we open the account as the owner `0x6f0f6DA1852857d7789f68a28bba866671f3880D`.
```

Comment:
@jpop32 I added this sentence, does it look right to you?

Reply:
Yep, makes sense to note that, thanks.